tensornetwork/network_components.py

```diff
@@ -1011,9 +1010,35 @@ def axis2(self, axis2: int) -> None:
           'Edge has been disabled, setting node1 is no longer possible')
     self._axes[1] = axis2
 
+  @property
+  def axes(self) -> Tuple[int]:
+    if self.is_disabled:
+      raise ValueError(
+          'Edge has been disabled, accessing axes is no longer possible')
+    return tuple(self._axes)
+
+  @axes.setter
+  def axes(self, axes: Iterable[int]) -> None:
+    if self.is_disabled:
+      raise ValueError(
+          'Edge has been disabled, setting axes is no longer possible')
+    if not all(map(lambda x: isinstance(x, int) or x is None), axes):
+      raise TypeError("All axis values should be int type")
+    self._axes = list(axes)
+
+  def set_axis(self, axis_pos, axis) -> None:
+    if axis_pos < 0 or axis_pos >= len(self._axes):
+      raise ValueError(
+          f"axis_pos {axis_pos} is outside the range 0 - {len(self._axes) - 1}"
+          )
+    if not (isinstance(axis, int) or axis is None):
+      raise TypeError("axis must be of type int")
+    self._axes[axis_pos] = axis
+
+
```

Comment:
Change node axes to a tuple returning property and a setter that takes an iterable and checks type

Reply:
A single value update can be done with the set_axis